app/uk/gov/hmrc/hec/services/TaxCheckService.scala

```diff
@@ -105,4 +111,17 @@ class TaxCheckServiceImpl @Inject() (
       HECTaxCheckMatchResult(taxCheckMatchRequest, timeProvider.currentDateTime, HECTaxCheckMatchStatus.NoMatch)
   }
 
+  def getUnexpiredTaxCheckCodes(
+    ggCredId: GGCredId,
+    today: LocalDate
+  )(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, Error, List[TaxCheckCodeListItem]] =
+    taxCheckStore
+      .getTaxCheckCodes(ggCredId)
+      .map(
+        _.filter(item => item.expiresAfter.isEqual(today) || item.expiresAfter.isAfter(today))
```

Comment:
if `expiresAfter` is equal to today then the tax check code is unexpired - `expiresAfter` was meant in the sense that it expires when today is after the specified date

Reply:
yeah, that's exactly what it does, the filter is keeping only the valid tax check codes.

Reply:
oh yeah, sorry - whoops :laughing: